### DISCUSSION TITLE
Can't render display in PikaOs on and igpu with drivers under Proxmox with igpu passthrough

### DISCUSSION THREAD

**Discussion by mkuznetsov** [LizardByte/community#421] [Sunshine] [answered]
### Is there an existing issue for this?

- [X] I have searched the existing issues

### Is your issue described in the documentation?

- [X] I have read the documentation

### Is your issue present in the nightly release?

- [X] This issue is present in the nightly release

### Describe the Bug

I have igpu AMD 610M, installed drivers under PikaOs (ubuntu with game optimizations) VM under Proxmox. I use Sunshine on same server with Windows VMs without problems. One time while I connected via vnc it worked but very slow. When I disable vnc session and server and tried to connected directly. I have monitor connected to HDMI and dummy plug. With both I get error of not able to detect display.
Installed sunshine through Flatpak.  How can I start while startup?

I tried command with setcap. same result.  

### Expected Behavior

It open session with display or dummy plug connected, same as on Windows VMs 
If amd igpu drivers failed than use software encoding. But software encoding is not available. 

### Additional Context

01:00.0 VGA compatible controller: Advanced Micro Devices, Inc. [AMD/ATI] Raphael (rev da) (prog-if 00 [VGA controller])
	Subsystem: Advanced Micro Devices, Inc. [AMD/ATI] Raphael
	Physical Slot: 0
	Control: I/O+ Mem+ BusMaster- SpecCycle- MemWINV- VGASnoop- ParErr- Stepping- SERR- FastB2B- DisINTx-
	Status: Cap+ 66MHz- UDF- FastB2B- ParErr- DEVSEL=fast >TAbort- <TAbort- <MAbort- >SERR- <PERR- INTx-
	Interrupt: pin A routed to IRQ 16
	Region 0: Memory at 380000000000 (64-bit, prefetchable) [size=256M]
	Region 2: Memory at 380010000000 (64-bit, prefetchable) [size=2M]
	Region 4: I/O ports at 8000 [size=256]
	Region 5: Memory at 81000000 (32-bit, non-prefetchable) [size=512K]
	Expansion ROM at 81080000 [disabled] [size=64K]
	Capabilities: [48] Vendor Specific Information: Len=08 <?>
	Capabilities: [50] Power Management version 3
		Flags: PMEClk- DSI- D1- D2- AuxCurrent=0mA PME(D0-,D1-,D2-,D3hot-,D3cold-)
		Status: D0 NoSoftRst- PME-Enable- DSel=0 DScale=0 PME-
	Capabilities: [64] Express (v2) Legacy Endpoint, MSI 00
		DevCap:	MaxPayload 256 bytes, PhantFunc 0, Latency L0s <4us, L1 unlimited
			ExtTag+ AttnBtn- AttnInd- PwrInd- RBE+ FLReset-
		DevCtl:	CorrErr- NonFatalErr- FatalErr- UnsupReq-
			RlxdOrd+ ExtTag+ PhantFunc- AuxPwr- NoSnoop+
			MaxPayload 128 bytes, MaxReadReq 512 bytes
		DevSta:	CorrErr- NonFatalErr- FatalErr- UnsupReq- AuxPwr- TransPend-
		LnkCap:	Port #0, Speed 16GT/s, Width x16, ASPM L0s L1, Exit Latency L0s <64ns, L1 <1us
			ClockPM- Surprise- LLActRep- BwNot- ASPMOptComp+
		LnkCtl:	ASPM Disabled; RCB 64 bytes, Disabled- CommClk+
			ExtSynch- ClockPM- AutWidDis- BWInt- AutBWInt-
		LnkSta:	Speed 16GT/s, Width x16
			TrErr- Train- SlotClk+ DLActive- BWMgmt- ABWMgmt-
		DevCap2: Completion Timeout: Range ABCD, TimeoutDis+ NROPrPrP- LTR-
			 10BitTagComp+ 10BitTagReq- OBFF Not Supported, ExtFmt+ EETLPPrefix+, MaxEETLPPrefixes 1
			 EmergencyPowerReduction Not Supported, EmergencyPowerReductionInit-
			 FRS-
			 AtomicOpsCap: 32bit+ 64bit+ 128bitCAS-
		DevCtl2: Completion Timeout: 50us to 50ms, TimeoutDis- LTR- 10BitTagReq- OBFF Disabled,
			 AtomicOpsCtl: ReqEn+
		LnkCap2: Supported Link Speeds: 2.5-16GT/s, Crosslink- Retimer+ 2Retimers+ DRS-
		LnkCtl2: Target Link Speed: 16GT/s, EnterCompliance- SpeedDis-
			 Transmit Margin: Normal Operating Range, EnterModifiedCompliance- ComplianceSOS-
			 Compliance Preset/De-emphasis: -6dB de-emphasis, 0dB preshoot
		LnkSta2: Current De-emphasis Level: -3.5dB, EqualizationComplete+ EqualizationPhase1+
			 EqualizationPhase2+ EqualizationPhase3+ LinkEqualizationRequest-
			 Retimer- 2Retimers- CrosslinkRes: unsupported
	Capabilities: [a0] MSI: Enable- Count=1/4 Maskable- 64bit+
		Address: 0000000000000000  Data: 0000
	Capabilities: [c0] MSI-X: Enable- Count=4 Masked-
		Vector table: BAR=5 offset=00042000
		PBA: BAR=5 offset=00043000
	Capabilities: [100 v1] Vendor Specific Information: ID=0001 Rev=1 Len=010 <?>
	Capabilities: [2a0 v1] Access Control Services
		ACSCap:	SrcValid- TransBlk- ReqRedir- CmpltRedir- UpstreamFwd- EgressCtrl- DirectTrans-
		ACSCtl:	SrcValid- TransBlk- ReqRedir- CmpltRedir- UpstreamFwd- EgressCtrl- DirectTrans-
	Capabilities: [2b0 v1] Address Translation Service (ATS)
		ATSCap:	Invalidate Queue Depth: 00
		ATSCtl:	Enable+, Smallest Translation Unit: 00
	Capabilities: [2c0 v1] Page Request Interface (PRI)
		PRICtl: Enable- Reset-
		PRISta: RF- UPRGI- Stopped+
		Page Request Capacity: 00000100, Page Request Allocation: 00000000
	Kernel modules: amdgpu

### Host Operating System

Linux

### Operating System Version

PikaOs on 6.8.3

### Architecture

64 bit

### Sunshine commit or version

0.22

### Package

Linux - flatpak

### GPU Type

AMD

### GPU Model

AMD 610M

### GPU Driver/Mesa Version

24.01

### Capture Method (Linux Only)

default

### Config

```Shell
default
```

### Apps

```json
desktop
```

### Relevant log output

```shell
[2024:04:04:14:45:59]: Info: Avahi service Sunshine successfully established.
[2024:04:04:14:46:21]: Info: // Testing for available encoders, this may generate errors. You can safely ignore those errors. //
[2024:04:04:14:46:21]: Info: Trying encoder [nvenc]
[2024:04:04:14:46:22]: Info: Encoder [nvenc] failed
[2024:04:04:14:46:22]: Info: Trying encoder [vaapi]
[2024:04:04:14:46:22]: Info: Encoder [vaapi] failed
[2024:04:04:14:46:22]: Info: Trying encoder [software]
[2024:04:04:14:46:23]: Info: Encoder [software] failed
[2024:04:04:14:46:23]: Fatal: Unable to find display or encoder during startup.
[2024:04:04:14:46:23]: Fatal: Please check that a display is connected and powered on.
```

**Comment by ReenigneArcher** [comment, marked as answer]
For software encoding you can use [Xvfb](https://en.wikipedia.org/wiki/Xvfb) to create a virtual display. I don't know how well this will work, but we use it in our unit tests to verify the software encoder passes the basic tests.
